### ISSUE TITLE
Swift 5.9 : unit tests fail with linker error

### ISSUE THREAD

**Issue by sebsto** [sebsto/xcodeinstall#33]
**Describe the bug**
When the root package manifest declares a tools version of at least 5.9, the commands `swift test` or `swift build --build-test` fails with linker error for symbols that are a dependency of the main project.

This is very similar to what is decsribed here, except that I can reproduce this on macOS 14.1 too
https://github.com/apple/swift-package-manager/issues/6940#issuecomment-1728597937

**To Reproduce**
```
# be sure to have Swift 5.9 installed 
git clone https://github.com/sebsto/xcodeinstall.git 
cd xcodeinstall 
# change tool version to 5.9 in `Package.swift`
swift test
```

**Detailed error messages**
```
Building for debugging...
ld: Undefined symbols:
  ArgumentParser.ArgumentVisibility.default.unsafeMutableAddressor : ArgumentParser.ArgumentVisibility, referenced from:
      default argument 1 of ArgumentParser.OptionGroup.init(title: Swift.String, visibility: ArgumentParser.ArgumentVisibility) -> ArgumentParser.OptionGroup<A> in CLIAuthenticate.swift.o
  ArgumentParser.ArgumentHelp.init(stringLiteral: Swift.String) -> ArgumentParser.ArgumentHelp, referenced from:
      variable initialization expression of xcodeinstall.MainCommand.Download.(_name in _EB4AB93595B2E1C16A7F4A6D2B09C29A) : ArgumentParser.Option<Swift.String?> in CLIDownload.swift.o
      xcodeinstall.MainCommand.Download.init() -> xcodeinstall.MainCommand.Download in CLIDownload.swift.o
      xcodeinstall.MainCommand.Download.init(from: Swift.Decoder) throws -> xcodeinstall.MainCommand.Download in CLIDownload.swift.o
      default argument 3 of xcodeinstall.MainCommand.Download.init(globalOptions: ArgumentParser.OptionGroup<xcodeinstall.MainCommand.GlobalOptions>, downloadListOptions: ArgumentParser.OptionGroup<xcodeinstall.MainCommand.DownloadListOptions>, cloudOption: ArgumentParser.OptionGroup<xcodeinstall.MainCommand.CloudOptions>, name: ArgumentParser.Option<Swift.String?>) -> xcodeinstall.MainCommand.Download in CLIDownload.swift.o
      variable initialization expression of xcodeinstall.MainCommand.Install.(_name in _48DC8EDB848BB62F56E00CA5D15CEE92) : ArgumentParser.Option<Swift.String?> in CLIInstall.swift.o
      xcodeinstall.MainCommand.Install.init() -> xcodeinstall.MainCommand.Install in CLIInstall.swift.o
      xcodeinstall.MainCommand.Install.init(from: Swift.Decoder) throws -> xcodeinstall.MainCommand.Install in CLIInstall.swift.o
      ...
```

**Workarounds**

There are two workaround I found 

1. Revert back tool version to 5.8 in `Package.swift`

- or - 

2. use `xcodebuild test -scheme xcodeinstall` to run the unit tests.  When run by Xcode (command line or GUI), there is no linker error. But this is a macOS only workaround.

**Comment by sebsto** [comment]
I found another workaround : Install Xcode 15.1 beta 2.
This comes with Swift 5.9.2 and `swift test` succeed with swift tools 5.9 in the package manifest

**Comment by sebsto** [comment]
This is fixed with swift 5.9.2 that comes with Xcode 15.1

Closing
